roles/edpm_ovn/molecule/default/verify-tasks.yml

```diff
@@ -0,0 +1,96 @@
+# we have to use shell to extract the values because openvswitch ansible
+# module doesn't seem to support dashes in key names, see:
+# https://github.com/ansible-collections/openvswitch.openvswitch/issues/111
+- name: Verify ovn-remote
+  ansible.builtin.shell: >
+    /usr/bin/ovs-vsctl get open_vswitch . external_ids:ovn-remote
+  register: output
+  failed_when: output.stdout != '"tcp:1.2.3.4:5678"'
+
+- name: Verify ovn-encap-type
+  ansible.builtin.shell: >
+    /usr/bin/ovs-vsctl get open_vswitch . external_ids:ovn-encap-type
+  register: output
+  failed_when: output.stdout != 'geneve'
+
+- name: Verify ovn firewall rules are applied
+  block:
+    - name: Rule source file exists
+      become: true
+      ansible.builtin.stat:
+        path: "/var/lib/edpm-config/firewall/ovn.yaml"
+      register: rule_source_exists
+
+    - name: Assert ovn rule source file exists
+      ansible.builtin.assert:
+        that:
+          - rule_source_exists.stat.exists
+        fail_msg: "ovn rules source file does not exist"
+
+    - name: Check if nftables rule exists in /etc/nftables/edpm-rules.nft
+      become: true
+      ansible.builtin.shell: grep -q "{{ item }}" /etc/nftables/edpm-rules.nft
+      register: rule_exists
+      ignore_errors: true
+      loop:
+        - "118 neutron vxlan networks"
+        - "119 neutron geneve networks"
+        - "120 neutron geneve networks no conntrack"
+        - "121 neutron geneve networks no conntrack"
```

Comment:
I wonder if wouldn't it be better to move to testinfra as verifier. It has easy to used dedicated modules to e.g. test if file exists and if it contains some strings, see https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.modules.file.File
I'm using it now in the edpm_neutron_{metadata,dhcp} roles to test sidecars

Reply:
Perhaps though here I'm just moving code around.